symtabAPI/src/emitElf.C

```diff
@@ -1990,6 +1990,24 @@ bool emitElf<ElfTypes>::createSymbolTables(set<Symbol *> &allSymbols) {
     if (!obj->getAllNewRegions(newSecs))
         log_elferror(err_func_, "No new sections to add");
 
+    for (unsigned long nsi = 0; nsi < newSecs.size(); nsi++) {
+	int sec_addr, prev_size;
+	if (newSecs[nsi]->getDiskOffset())
+	  sec_addr = newSecs[nsi]->getDiskOffset() + library_adjust;
+	else
+	  sec_addr += prev_size;
+	prev_size = newSecs[nsi]->getDiskSize();
+	if (".dynamic" == newSecs[nsi]->getRegionName()) {
```

Comment:
Hard-coded strings make me nervous. Are '.dynamic' and '_DYNAMIC' guaranteed by the ELF standard? Has this been tested on binaries created by clang?

Reply:
#1176 actually fixes the static executable problem so how about we punt on this since it is not essential to fix the problem at hand.

Reply:
@hainest  I found the reference in the elf spec:
[elf spec]( https://refspecs.linuxfoundation.org/elf/elf.pdf)
If an object file participates in dynamic linking, its program header table will have an element of type
PT_DYNAMIC. This ‘‘segment’’ contains the .dynamic section. A special symbol, _DYNAMIC, labels the
section, which contains an array of the following structures.

Reply:
All good, then. I just wanted to double-check.